File: anagram_palindrome.py
def is_anagram_of_palindrome(exp):
    """ Is the word of an anagram of a palindrome?"""
    adict = {}
    for char in exp:
        if char in adict:
            adict[char] += 1
        else:
            adict[char] = 1
    dict_values = adict.values()
    odds = []
    for val in dict_values:
      if val % 2 != 0:
        odds.append(val)
    return len(odds) <= 1
print is_anagram_of_palindrome('racecar')


def anagram_of_palindrome(exp):
    """Use a set."""
    anagram_set = set()
    for char in exp:
        if char in anagram_set:
            anagram_set.remove(char)
        else:
            anagram_set.add(char)
    return len(anagram_set) < 2
    #if length of anagram set is greater than two, the word is not a palindrome.
    #if length of anagram set is 1, it was an odd-length word
    #if length of anagram set is 0, it was an even-length word!
print anagram_of_palindrome('red')
print anagram_of_palindrome('tacocat')

File: anagram_palindrome.rb
# def is_anagram_of_palindrome(exp)
#     adict = {}
#     for char in exp:
#         if char in adict:
#             adict[char] += 1
#         else:
#             adict[char] = 1
#     dict_values = adict.values()
#     odds = []
#     for val in dict_values:
#       if val % 2 != 0:
#         odds.append(val)
#     return len(odds) <= 1
# print is_anagram_of_palindrome('racecar')

def anagram_of_palindrome?(text)
  dict = {}

  text.each_char do |word|
    if dict.has_key? word
      dict[word] += 1
    else
      dict[word] = 1
    end
  end
puts dict
values = dict.values
output = []
values.map {|num| output.push(num) if num.odd?}
return output.length <= 1
end

puts anagram_of_palindrome("poop42")
